### PR TITLE
Add animated gif support to notifications, comments, and my site comments

### LINKED ISSUE #17961: Gifs blocks aren't showing up in the notifications or comments
### Expected behavior
When using a gif block the image should be rendered correctly

### Actual behavior
The block is not rendered

### Steps to reproduce the behavior
1. Receive a comment reply notification whose content is a Gif block
2. Open the notifications tab
3. Open the notification
4. The gif block will not be rendered

##### Tested on [device], iOS [version], WPiOS [version]
iPhone 11 Pro, iOS 15, WPiOS 19.1

### LINKED ISSUE #17962: Gifs don't play in comments
### Expected behavior
When embedding a gif it should play

### Actual behavior
The gif does not play

### Steps to reproduce the behavior
1. Receive a comment reply that contains an image block that is a gif
2. Open the Notifications tab
3. Tap on the comment reply notification
4. Notice the embedded gif does not play

##### Tested on [device], iOS [version], WPiOS [version]

iPhone 11 Pro, iOS 15, WPiOS 19.1

### PR DESCRIPTION
**Pull request by emilylaguna**
Fixes: #17962
Fixes: #17961

https://user-images.githubusercontent.com/793774/154581024-59770e92-b4a9-4e0f-9fe5-d5505fb94226.mp4

### Description
Adds animated gif support for the Notifications, Reader Comments, and My Site Comments.

This uses an iOS 15 class `NSTextAttachmentViewProvider` that allows us to provide a custom view for `NSTextAttachment`'s

The new subclass `AnimatedGifAttachmentViewProvider` is added as a handler for kUTTypeGIF types and provides a `CachedAnimatedImageView` to allow the gif to play.

Some reworking was required to get this to work correctly since the images are downloaded as an `RCTAnimatedImage` and then resized down to a single frame `UIImage`. To work around this I added a new wrapper class `AnimatedImageWrapper` that allows me to read the original gifData and then later pass that to the `CachedAnimatedImageView`. This doesn't allow gifs that are too big (over 50mb currently) to play and will instead show a "single frame" image.

Some of these fixes are currently iOS 15 and above, when I looked at the data the majority of users are on some version of iOS 15 anyways.

### To test:
Note: You will need 2 accounts to receive the comment notifications.

#### Image Block
1. Go to a test blog that has blocks enabled for comments (P2 sites work well) with your second account (the one that isn't logged into the app)
2. Reply to a comment and add an image block
3. Upload a gif image to the block and leave your comment
4. Receive the notification in the app
5. Tap on it
6. 👁️  Verify the gif plays correctly 
7. Tap on the header to view the rest of the gifs and verify the gif plays correctly there
8. Tap on the header view again, then tap on your site URL to open the Reader Site View
9. Tap on the post that has the gif comment
10. At the bottom of the reader view, verify you see the gif in the comment preview section
11. 👁️ Verify the gif is animating

#### Gif Block
1. Repeat the steps above but instead of adding an image block with a gif, add a gif block and search for a gif to use
2. Verify D75182-code is deployed. If it's not follow the testing steps there

## Regression Notes
1. Potential unintended areas of impact
- Tapping on Edit Image in the Gutenberg editor uses the ImageDownloader class that I modified

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by mokagio** [comment]
@emilylaguna I can see you explicitly asked for this to be part of 19.3:

 Screen Shot 2022-02-21 at 1 50 34 pm

However, this PR hasn't been review yet and I'm starting the code freeze process soon, so this won't make it into the first beta. I'm happy to ship a new beta once this one is approved. Or, we can bump this to 19.4. Up to you.

- If this should stay in 19.3, we'll need to update the base branch for the PR from `trunk` to `release/19.3`;
- Otherwise, we simply need to update the milestone to 19.4

Can you please take the action you find more appropriate? Thanks

cc @momo-ozawa and @dvdchr, the current reviewers, to avoid this being merged before Emily replies.

Thanks y'all!

**Comment by emilylaguna** [comment]
Thanks @mokagio this should be on 19.3, I'll rebase to `release/19.3` early this week.

**Comment by mokagio** [comment]
Thanks @emilylaguna 

> I'll rebase to `release/19.3` early this week.

I changed the base branch here just now. We now have conflicts on `RELEASE-NOTES.txt`, which was to be expected given the changes in that file made by the code freeze process. You wrote "rebase" and I'm not whether to read it as "change the base on GitHub" or "run `git rebase release/19.3`". As such, I won't fix this conflict and let you do as you prefer with this branch and PR.

Cheers.

**Review by momo-ozawa — COMMENTED** [review]
Tested both image blocks and gif blocks, and testing steps work as advertised. However, I ran into an issue while testing the gif block / D75182-code.

If I open a post with an **image block** gif comment from the Reader Site view, the gif in the comment preview section is not displayed.

**Comment by emilylaguna** [comment]
@momo-ozawa Sadly, that's a bug with that preview section for private sites. It doesn't make an authenticated call to get the images.

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,3 +1,7 @@
+19.4
+-----
+* [**] Adds animated Gif support in notifications and comments [#17981]
```
Comment (momo-ozawa) [inline RELEASE-NOTES.txt]: I see that this PR is pointing to 19.3 -- just a reminder that this needs to go into the 19.3 section

**Review by momo-ozawa — APPROVED** [review]
@emilylaguna 

> Sadly, that's a bug with that preview section for private sites. It doesn't make an authenticated call to get the images.

In that case, LGTM! Left 1 comment about release notes.

**Comment by mokagio** [comment]
@emilylaguna  this has been bundled as part of [19.3 beta 1 (19.3.0.1)](https://github.com/wordpress-mobile/WordPress-iOS/pull/18011).

Thanks for your work 🙌

**Pull request merged**